### DISCUSSION TITLE
documentaion?

### DISCUSSION THREAD

**Discussion by Blazzycrafter** [ppy/osu#18177] [Q&A]
exist a osu dev documentation?
where i can se as example 
how i can develope a mod?

because the one line after the contructors and usings 
```
public override Type[] IncompatibleMods => base.IncompatibleMods.Append(typeof(OsuModAutopilot)).ToArray();
```
in `osu-master\osu.Game.Rulesets.Osu\Mods\OsuModNoFail.cs` [directlink to repo](https://github.com/peppy/osu/blob/master/osu.Game.Rulesets.Osu/Mods/OsuModNoFail.cs)

says yeah this mods cant be used...
yea but where is the difficulty setted?

for this and more is a documentation good 
maybe i can programm this: [D\18176 (osu mod: autorestart)](https://github.com/ppy/osu/discussions/18176)

**Comment by frenzibyte** [comment]
I'm not entirely sure what you're asking here, sorry. Just going to mention that #18176 can wait until it's decided whether such mod is valid, before PR'ing an implementation for it.

**Reply by Blazzycrafter** [reply]
just asking if and where there is documentation?

**Reply by integerrr** [reply]
Putting #18176 aside, you should be looking at abstract implementation of mods instead of ones in the specific ruleset i.e. `ModNoFail` or `ModBlockFail` intstead of `OsuModNoFail`
